src/content/cookbook/architecture/result.md

```diff
@@ -0,0 +1,460 @@
+are usually composed of ViewModels, 
+Repositories, and Services, among other parts. 
+When a function in one of these components fails, 
+it should communicate the error to the calling component.
+
+Typically, that's done with exceptions. 
+For example, 
+an API client Service failing to communicate with the remote server
+might throw an HTTP Error Exception. 
+The calling component, 
+for example a Repository, 
+would have to either capture this exception 
+or ignore it and let the calling ViewModel handle it.
+
+This can be observed in the following example. Consider these classes:
+
+- A service named `ApiClientService` 
+that performs API calls to a remote service.
+- A repository named `UserProfileRepository` 
+that uses the `ApiClientService` to provide the `UserProfile`.
+- A ViewModel named `UserProfileViewModel` 
+that uses the `UserProfileRepository`.
+
+The `ApiClientService` contains a method named `getUserProfile`
+ that can throw exceptions in different situations:
+
+- The method throws an `HttpException` if the response code isn’t 200.
+- The JSON parsing method might throw an exception 
+if the response is not formatted correctly.
+- The Http Client might throw an exception due to networking issues.
+
+For example:
+
+<?code-excerpt "lib/no_result.dart (ApiClientService)"?>
+```dart
+class ApiClientService {
+  // ···
+
+  Future<UserProfile> getUserProfile() async {
+    try {
+      final request = await client.get(_host, _port, '/user');
+      final response = await request.close();
+      if (response.statusCode == 200) {
+        final stringData = await response.transform(utf8.decoder).join();
+        return UserProfile.fromJson(jsonDecode(stringData));
+      } else {
+        throw const HttpException('Invalid response');
+      }
+    } finally {
+      client.close();
+    }
+  }
+}
+```
+
+The `UserProfileRepository` doesn’t need to handle 
+the exceptions from the `ApiClientService`. 
+In this example, it just returns the value from the API Client.
+
+<?code-excerpt "lib/no_result.dart (UserProfileRepository)"?>
+```dart
+class UserProfileRepository {
+  // ···
+
+  Future<UserProfile> getUserProfile() async {
+    return await _apiClientService.getUserProfile();
+  }
+}
+```
+
+Finally, the `UserProfileViewModel` 
+should capture all exceptions and handle the errors.
+
+This can be done by wrapping 
+the call to the `UserProfileRepository` with a try-catch:
+
+<?code-excerpt "lib/no_result.dart (UserProfileViewModel)"?>
+```dart
+class UserProfileViewModel extends ChangeNotifier {
+  // ···
+
+  Future<void> load() async {
+    try {
+      _userProfile = await userProfileRepository.getUserProfile();
+      notifyListeners();
+    } on Exception catch (exception) {
+      // handle exception
+    }
+  }
+}
+```
+
+In reality, a developer might forget to properly capture exceptions and end up with the following code. It would compile and run, 
+but will crash if one of the exceptions mentioned previously occurs:
```

Comment:
```suggestion
In reality, a developer might forget to properly capture exceptions
and end up with the following code. It would compile and run, 
but will crash if one of the exceptions mentioned previously occurs:
```

Line length